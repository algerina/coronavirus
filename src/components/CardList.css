.cards {
    display: flex;
    flex-wrap: wrap;
    margin: 0.2rem;
}

