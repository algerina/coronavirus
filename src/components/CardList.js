import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Card from './Card';
import SearchBox from './SearchBox';
import Header from './Header';
import World from './World';
import { getCountriesAsync } from '../redux/reducers/dataReducer';

const CardList = () => {
    const [searchTerm, setSearchTerm] = useState('');

    useEffect(() => {
        dispatch(getCountriesAsync());
    }, []);

    const dispatch = useDispatch();
    let countries = useSelector((state) => state.dataReducer);

    // const searchChange = (e) => {
    //   const searchTerm = e.target.value;
  
    if (searchTerm.length > 0) {
      countries = countries.filter((country) => country.country.toLowerCase().includes(searchTerm.toLowerCase()));
    }

    //   return countries;
    // }

    return (
        <div>
            <Header />
            <SearchBox setSearchTerm={setSearchTerm} />
            <World />
            <div>
                {countries.map(country => {
                    if (country.countryInfo.iso3) {
                        return (
                            <Card
                                key={country.countryInfo.iso3}
                                id={country.countryInfo._id}
                                data={country}
                            />
                        );
                    }
                })}
            </div>
        </div>
    );
};
export default CardList;