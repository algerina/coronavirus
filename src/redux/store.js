import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import dataReducer from './reducers/dataReducer';

const reducer = combineReducers({ dataReducer });

const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);

export default store;
