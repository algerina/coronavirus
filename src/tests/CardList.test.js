import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import CardList from '../components/CardList';
import store from '../redux/store';

describe('Describe', () => {
  test('shoudl match snapShot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <CardList />
        </Router>
      </Provider>,
    ).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
