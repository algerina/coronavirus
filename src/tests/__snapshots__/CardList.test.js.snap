// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Describe shoudl match snapShot 1`] = `[Function]`;
